### ISSUE TITLE
HL_TRACE_FILE fails to create new file on Windows

### ISSUE THREAD

**Issue by Trass3r** [halide/Halide#927]
Tested with VS2015 and jit compilation.

```
HL_TRACE=3
HL_TRACE_FILE=D:\...\bla.bin

src\runtime\tracing.cpp:210 Assert failed: (fd > 0) && "Failed to open trace file\n"
```

The getenv("HL_TRACE_FILE") works correctly but the open doesn't for some reason.

**Comment by Trass3r** [comment]
I thought pmode would be the problem according to https://msdn.microsoft.com/en-us/library/z0kc8e3z.aspx but using _S_IWRITE still triggered the assertion.

Interestingly enough, when creating bla.bin manually up front it works and writes to that file. Though it gets deleted later on.

**Comment by ongjunjie** [comment]
This issue should be resolved by 50a3b2164afba0331bbea68c3f27753abcd87071

The hard coded values of ``O_CREAT`` and ``O_APPEND`` are interpreted as ``O_EXCL`` and ``O_TEMPORARY`` respectively on windows platform. This explains why:
  1. New files can't be created.
  2. Trace files are deleted when program exits.

**Comment by dsharletg** [comment]
This is probably resolved as mentioned above, please reopen if not.
